torch/csrc/inductor/aot_inductor_model_container.h

```diff
@@ -51,6 +76,49 @@ class AOTInductorModelContainer {
       output_dtypes_.push_back(model->get_output_dtype(i));
       max_output_shapes_.emplace_back(model->max_output_shape(i));
     }
+
+    size_t num_constants = model->num_constants();
+    std::vector<size_t> constants_internal_offset(num_constants);
+    // Compute required blob size with 64-alignment
+    size_t max_blob = 0;
+    for (size_t i = 0; i < num_constants; i++) {
+      size_t data_size = model->constant_data_size(i);
+      if (data_size % GPU_ALIGNMENT) {
+        data_size = GPU_ALIGNMENT + (data_size / GPU_ALIGNMENT) * GPU_ALIGNMENT;
+      }
+      constants_internal_offset[i] = max_blob;
+      max_blob += data_size;
+    }
+    constant_blob_ = RAII_cudaMalloc(max_blob);
+
+    constants_->reserve(num_constants);
+    auto* constants_ptr = static_cast<uint8_t*>(constant_blob_.get());
+    size_t bytes_read = 0;
+    for (size_t i = 0; i < num_constants; i++) {
+      std::string name = model->constant_name(i);
+      size_t data_size = model->constant_data_size(i);
+      auto* internal_ptr = constants_ptr + constants_internal_offset[i];
+      // Copy data to GPU memory
+      // TODO: Handle shared storage case.
+      C10_CUDA_CHECK(cudaMemcpy(
+          internal_ptr,
+          _binary_constants_bin_start + bytes_read,
+          data_size,
+          cudaMemcpyHostToDevice));
+      bytes_read += data_size;
+
+      // Create at::Tensor from copied memory.
+      auto dtype = model->constant_type(i);
+      auto size = model->constant_shape(i);
+      auto stride = model->constant_stride(i);
+      auto offset = model->constant_offset(i);
+
+      auto tensor = at::from_blob(
+          internal_ptr, size, stride, at::device(at::kCUDA).dtype(dtype));
+      tensor.unsafeGetTensorImpl()->set_sizes_and_strides(size, stride);
+      tensor.unsafeGetTensorImpl()->set_storage_offset(offset);
```

Comment:
Where do we write out the tensors? We should add a test case with a storage offset